src/main/java/de/tum/in/www1/artemis/service/connectors/hades/HadesCITriggerService.java

```diff
@@ -0,0 +1,173 @@
+            headers.add(HttpHeaders.AUTHORIZATION, "Basic " + Base64.getEncoder().encodeToString(((hadesUsername + ":" + hadesPassword).getBytes())));
+            headers.add(HttpHeaders.CONTENT_TYPE, "application/json");
+            HttpEntity<HadesBuildJobDTO> request = new HttpEntity<>(job, headers);
+
+            var response = restTemplate.postForEntity(hadesServerUrl + "/build", request, JsonNode.class);
+            log.debug("Hades response: {}", response);
+        }
+        catch (Exception e) {
+            log.error("Error triggering the build for {}", job, e);
+        }
+    }
+
+    private HadesBuildJobDTO createJob(ProgrammingExerciseParticipation participation) throws JsonProcessingException {
+
+        // Build Job Metadata
+        var metadata = new HashMap<String, String>();
+        metadata.put("PLAN_KEY", participation.getBuildPlanId());
+
+        var steps = new ArrayList<HadesBuildStepDTO>();
+
+        // Create Clone Step
+        // TODO: We need a solution to clone with less critical credentials - This requires changes in the localvc implementation
+        var cloneMetadata = new HashMap<String, String>();
+        cloneMetadata.put("REPOSITORY_DIR", "/shared");
+        cloneMetadata.put("HADES_TEST_USERNAME", gitUsername);
+        cloneMetadata.put("HADES_TEST_PASSWORD", gitPassword);
+        cloneMetadata.put("HADES_TEST_URL", participation.getProgrammingExercise().getTestRepositoryUri());
+        cloneMetadata.put("HADES_TEST_PATH", "./");
+        cloneMetadata.put("HADES_TEST_ORDER", "1");
+        cloneMetadata.put("HADES_ASSIGNMENT_USERNAME", gitUsername);
+        cloneMetadata.put("HADES_ASSIGNMENT_PASSWORD", gitPassword);
+        cloneMetadata.put("HADES_ASSIGNMENT_URL", participation.getVcsRepositoryUri().toString());
+        cloneMetadata.put("HADES_ASSIGNMENT_PATH", "./assignment");
+        cloneMetadata.put("HADES_ASSIGNMENT_ORDER", "2");
+        // TODO: Auxiliary Repository clone is not supported yet
+
+        steps.add(new HadesBuildStepDTO(1, "Clone", cloneDockerImage, cloneMetadata));
+
+        // Create Execute Step
+        var image = programmingLanguageConfiguration.getImage(participation.getProgrammingExercise().getProgrammingLanguage(),
+                Optional.ofNullable(participation.getProgrammingExercise().getProjectType()));
+        var script = buildScriptGenerationService.getScript(participation.getProgrammingExercise());
+        steps.add(new HadesBuildStepDTO(2, "Execute", image, script));
+
+        var resultMetadata = getStringStringHashMap(participation);
+        steps.add(new HadesBuildStepDTO(3, "Result", resultDockerImage, resultMetadata));
+
+        // Create Hades Job
+        var timestamp = java.time.Instant.now().toString();
+        return new HadesBuildJobDTO("Test", metadata, timestamp, 1, steps);
+    }
+
+    private HashMap<String, String> getStringStringHashMap(ProgrammingExerciseParticipation participation) {
```

Comment:
This is a pretty bad method name, as it does not say anything about its purpose.